### ISSUE TITLE
It just crashed while typing

### ISSUE THREAD

**Issue by bnbrown** [florisboard/florisboard#874]
~~~ 1620950684848.stacktrace ~~~

java.lang.OutOfMemoryError: Failed to allocate a 32 byte allocation with 1381088 free bytes and 1348KB until OOM, target footprint 268435456, growth limit 268435456; failed due to fragmentation (largest possible contiguous allocation 96206848 bytes)
	at java.util.ArrayList.iterator(ArrayList.java:834)
	at dev.patrickgold.florisboard.ime.nlp.NgramNode.listAllSameOrderWords(FlorisLanguageModel.kt:162)
	at dev.patrickgold.florisboard.ime.nlp.NgramNode.listAllSameOrderWords(FlorisLanguageModel.kt:163)
	at dev.patrickgold.florisboard.ime.nlp.NgramNode.listAllSameOrderWords(FlorisLanguageModel.kt:163)
	at dev.patrickgold.florisboard.ime.nlp.NgramNode.listAllSameOrderWords(FlorisLanguageModel.kt:163)
	at dev.patrickgold.florisboard.ime.nlp.NgramNode.listAllSameOrderWords(FlorisLanguageModel.kt:163)
	at dev.patrickgold.florisboard.ime.nlp.NgramNode.listAllSameOrderWords(FlorisLanguageModel.kt:163)
	at dev.patrickgold.florisboard.ime.nlp.NgramNode.listAllSameOrderWords(FlorisLanguageModel.kt:163)
	at dev.patrickgold.florisboard.ime.nlp.NgramNode.listAllSameOrderWords(FlorisLanguageModel.kt:163)
	at dev.patrickgold.florisboard.ime.nlp.FlorisLanguageModel.matchAllNgrams(FlorisLanguageModel.kt:253)
	at dev.patrickgold.florisboard.ime.dictionary.Flictionary.getTokenPredictions(Flictionary.kt:308)
	at dev.patrickgold.florisboard.ime.text.TextInputManager$onUpdateSelection$$inlined$let$lambda$1.invokeSuspend(TextInputManager.kt:372)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
~~~ 1621002764029.stacktrace ~~~

java.lang.OutOfMemoryError: Failed to allocate a 32 byte allocation with 95936 free bytes and 93KB until OOM, target footprint 268435456, growth limit 268435456; failed due to fragmentation (largest possible contiguous allocation 116129792 bytes)
	at java.util.LinkedHashMap.newNode(LinkedHashMap.java:280)
	at java.util.HashMap.putVal(HashMap.java:630)
	at java.util.HashMap.put(HashMap.java:611)
	at org.json.JSONObject.put(JSONObject.java:273)
	at org.json.JSONTokener.readObject(JSONTokener.java:394)
	at org.json.JSONTokener.nextValue(JSONTokener.java:104)
	at org.json.JSONObject.<init>(JSONObject.java:165)
	at org.json.JSONObject.<init>(JSONObject.java:182)
	at dev.patrickgold.florisboard.ime.text.gestures.GlideTypingManager$setWordData$1.invokeSuspend(GlideTypingManager.kt:79)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)

**Comment by Glitchy-Tozier** [comment]
Hi and thank you for taking the time to fill out this bug-report!

Could you please share which version of the app you are using?

**Comment by Glitchy-Tozier** [comment]
I'm assuming that this happened with v0.3.10, partially due to the timeframe, partially due to the log being formatted in the old way. This issue should be fixed by now (see #677)

If the crash should happen again, please open a new issue for it!
